ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutorUtil.java

```diff
@@ -0,0 +1,234 @@
+    );
+
+    assertThat("test did not generate any queries.", !queries.isEmpty());
+    return queries;
+  }
+
+  /**
+   * @param srClient if supplied, then schemas can be inferred from the schema registry.
+   */
+  private static List<PersistentQueryAndSortedSources> execute(
+      final KsqlEngine engine,
+      final String sql,
+      final KsqlConfig ksqlConfig,
+      final Map<String, Object> overriddenProperties,
+      final Optional<SchemaRegistryClient> srClient
+  ) {
+    final List<ParsedStatement> statements = engine.parse(sql);
+
+    final Optional<DefaultSchemaInjector> schemaInjector = srClient
+        .map(SchemaRegistryTopicSchemaSupplier::new)
+        .map(DefaultSchemaInjector::new);
+
+    return statements.stream()
+        .map(stmt -> execute(engine, stmt, ksqlConfig, overriddenProperties, schemaInjector))
+        .filter(queryAndSortedSources -> queryAndSortedSources.getSources() != null)
+        .map(
+            queryAndSortedSources -> new PersistentQueryAndSortedSources(
+                (PersistentQueryMetadata) queryAndSortedSources.getExecuteResult().getQuery().get(),
+                queryAndSortedSources.getSources()
+            ))
+        .collect(Collectors.toList());
+  }
+
+
+  @SuppressWarnings({"rawtypes","unchecked"})
+  private static QueryAndSortedSources execute(
+      final KsqlExecutionContext executionContext,
+      final ParsedStatement stmt,
+      final KsqlConfig ksqlConfig,
+      final Map<String, Object> overriddenProperties,
+      final Optional<DefaultSchemaInjector> schemaInjector
+  ) {
+    final PreparedStatement<?> prepared = executionContext.prepare(stmt);
+    final ConfiguredStatement<?> configured = ConfiguredStatement.of(
+        prepared, overriddenProperties, ksqlConfig);
+    final ConfiguredStatement<?> withSchema =
+        schemaInjector
+            .map(injector -> injector.inject(configured))
+            .orElse((ConfiguredStatement) configured);
+    final ExecuteResult executeResult = executionContext.execute(withSchema);
+    if (prepared.getStatement() instanceof CreateAsSelect) {
+      return new QueryAndSortedSources(
+          executeResult,
+          getSortedSources(
+              (CreateAsSelect)prepared.getStatement(),
+              executionContext.getMetaStore()));
+    }
+    return new QueryAndSortedSources(executeResult, null);
+  }
+
+  private static List<DataSource> getSortedSources(
+      final CreateAsSelect createAsSelect,
+      final MetaStore metaStore) {
+    final Relation from = createAsSelect.getQuery().getFrom();
+    if (from instanceof Join) {
+      final Join join = (Join) from;
+      final AliasedRelation left = (AliasedRelation) join.getLeft();
+      final AliasedRelation right = (AliasedRelation) join.getRight();
+      if (metaStore.getSource(left.getRelation().toString()) == null) {
+        throw new KsqlException("Sourde does not exist: " + left.getRelation().toString());
```

Comment:
```suggestion
        throw new KsqlException("Source does not exist: " + left.getRelation().toString());
```
Also two more instances of this typo below.